src/utils.ts

```diff
@@ -1,9 +1,15 @@
 import { useEffect, useCallback, useRef } from "react";
 
-export function useOnClickOutside(ref, handler) {
+type ClickOutsideCallback = (event: MouseEvent) => void;
+
+export function useOnClickOutside<T>(
```

Comment:
I think you could have this as `T extends HTMLElement` or something along those line to constrain it so that `refl.current` has attributes like `contains`